pkg/controllers/user/helm/common/common.go

```diff
@@ -111,6 +108,48 @@ func InstallCharts(rootDir, port string, obj *v3.App) error {
 	return nil
 }
 
+func GenerateAnswerSetValues(app *v3.App, tempDir string) ([]string, error) {
+	setValues := []string{}
+	// a user-supplied values file will overridden default values.yaml
+	if app.Spec.ValuesYaml != "" {
+		valuesYaml := filepath.Join(tempDir, "answers.yaml")
+		if err := ioutil.WriteFile(valuesYaml, []byte(app.Spec.ValuesYaml), 0755); err != nil {
+			return setValues, err
+		}
+		setValues = append([]string{"--values"}, valuesYaml)
+	}
+	// user-supplied values file will be overridden by --set parameters
```

Comment:
One thing comes to my mind that `values.yml` and `--set` parameters are against each other. Should we alert the users when they try to input both of them?

Reply:
not exactly, in `helm` user are allowed to use both `--set` and `--values` of user-supplied `values.yaml`, the user-supplied `values.yaml` will be overridden by the `--set` values. 

In Rancher we need to deal with two types of inputs, one is for the `cli` and the other one is for the UI, in the `cli` we should allow both `--set` and custom `values.yaml`, in the `UI` it is preferred to pass only one type of values either from answers for `--set` purpose or pass user-supplied `values.yaml`. 

the backend should be able to handle both `--values` and `--set` cases.